packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx

```diff
@@ -672,6 +675,16 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
       ev.preventDefault();
       ev.stopPropagation();
     }
+
+    if (!(ev.altKey || ev.metaKey) && (isUp || isDown)) {
+      if (this.state.menuProps) {
+        const menuProps = { ...this.state.menuProps, shouldFocusOnMount: true };
+        this.setState({ menuProps: menuProps });
```

Comment:
I know this component is already doing this, but storing props objects in state this way is very gross and I wanted to call it out as something we shouldn't do. 

Might want to consider using [setState to read and update state](https://reactjs.org/docs/faq-state.html#how-do-i-update-state-with-values-that-depend-on-the-current-state) rather than reading `this.state`.

Minor, but this can just be:
```tsx 
this.setState({ menuProps });
```